packages/components/pie-webc-core/src/test/forms/pie-form-manager.spec.ts

```diff
@@ -0,0 +1,191 @@
+        it('correctly sets up a submit event listener when a form is added', () => {
+            // Arrange
+            document.body.innerHTML = `
+                <form id="testForm"></form>
+            `;
+
+            const testForm = document.getElementById('testForm') as HTMLFormElement;
+            const manager = new PieFormManager();
+            const addEventListenerSpy = vi.spyOn(testForm, 'addEventListener');
+
+            // Act
+            manager.addForm(testForm);
+
+            // Assert
+            expect(addEventListenerSpy).toHaveBeenCalledWith('submit', expect.any(Function));
+        });
+
+        it('does not throw an error when trying to add the same form multiple times', () => {
+            // Arrange
+            document.body.innerHTML = `
+                <form id="testForm"></form>
+            `;
+
+            const testForm = document.getElementById('testForm') as HTMLFormElement;
+            const manager = new PieFormManager();
+
+            // Act
+            const addForm = () => manager.addForm(testForm);
+
+            // Assert
+            expect(addForm).not.toThrow();
+        });
+    });
+
+    describe('getForm', () => {
+        it('returns the PieFormData object associated with a form', () => {
+            // Arrange
+            document.body.innerHTML = `
+                <form id="testForm"></form>
+            `;
+
+            const testForm = document.getElementById('testForm') as HTMLFormElement;
+            const manager = new PieFormManager();
+            manager.addForm(testForm);
+
+            // Act
+            const result = manager.getForm(testForm);
+
+            // Assert
+            expect(result?.form).toStrictEqual(testForm);
+        });
+
+        it('returns undefined when a form is not in the manager', () => {
+            // Arrange
+            document.body.innerHTML = `
+                <form id="testForm"></form>
+            `;
+
+            const testForm = document.getElementById('testForm') as HTMLFormElement;
+            const manager = new PieFormManager();
+
+            // Act
+            const result = manager.getForm(testForm);
+
+            // Assert
+            expect(result).toBeUndefined();
+        });
+
+        it('returns undefined when a form has been deleted', () => {
+            // Arrange
+            document.body.innerHTML = `
+                <form id="testForm"></form>
+            `;
+
+            const testForm = document.getElementById('testForm') as HTMLFormElement;
+            const manager = new PieFormManager();
+            manager.addForm(testForm);
+            manager.deleteForm(testForm);
+
+            // Act
+            const result = manager.getForm(testForm);
+
+            // Assert
+            expect(result).toBeUndefined();
+        });
+    });
+
+    describe('deleteForm', () => {
+        it('does not return the PieFormData object associated with a form after it has been deleted', () => {
```

Comment:
I think this test is identical to the one above?

Reply:
correct! removing it now